offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/DependencyManager.groovy

```diff
@@ -236,436 +181,405 @@ class DependencyManager {
-        }
-
-        try {
-            this.maintainProductsViewModelArchive = new MaintainProductsViewModel(productsResourcesService, productUpdateEvent)
-        }catch (Exception e) {
-            log.error("Unexpected exception during ${MaintainProductsViewModel.getSimpleName()} view model setup.", e)
-        }
-
-        try {
-            this.createProductViewModel = new CreateProductViewModel()
-        }catch (Exception e) {
-            log.error("Unexpected exception during ${CreateProductViewModel.getSimpleName()} view model setup.", e)
-        }
-
-        try {
-            this.copyProductViewModel = new CopyProductViewModel(productUpdateEvent)
-        }catch (Exception e) {
-            log.error("Unexpected exception during ${CopyProductViewModel.getSimpleName()} view model setup.", e)
-        }
-
-        try {
-            this.searchAffiliationViewModel = new SearchAffiliationViewModel(affiliationService)
-        } catch (Exception e) {
-            log.error("Unexpected exception during ${SearchAffiliationViewModel.getSimpleName()} creation.")
-            log.debug("Unexpected exception during ${SearchAffiliationViewModel.getSimpleName()} creation.", e)
-        }
+    private AppView setupAppView() {
+        CreateAffiliationView createAffiliationView = createCreateAffiliationView(viewModel, affiliationService, customerDbConnector)
+        CreateOfferView createOfferView = createCreateOfferView(affiliationService, customerResourceService, personResourceService, managerResourceService, productsResourcesService, offerService, viewModel, customerDbConnector, customerDbConnector, offerDbConnector, offerDbConnector)
+        CreatePersonView createPersonView = createCreatePersonView(viewModel, affiliationService, customerResourceService, personResourceService, managerResourceService, customerDbConnector, customerDbConnector)
+        CreateOfferView updateOfferView = createUpdateOfferView(viewModel, affiliationService, customerResourceService, offerService, personResourceService, managerResourceService, productsResourcesService, offerUpdateEvent, customerDbConnector, offerDbConnector, customerDbConnector, offerDbConnector)
+        MaintainProductsView maintainProductsView = createMaintainProductsView(viewModel, productsResourcesService, productsDbConnector, productsDbConnector, productsDbConnector)
+        OfferOverviewView overviewView = createOfferOverviewView(viewModel, overviewService, projectResourceService, projectSpaceResourceService, offerUpdateEvent, projectCreatedEvent, projectMainConnector, projectMainConnector, offerDbConnector)
+        SearchAffiliationView searchAffiliationView = createSearchAffiliationView(affiliationService)
+        SearchPersonView searchPersonView = createSearchPersonView(viewModel, affiliationService, customerResourceService, managerResourceService, personResourceService, customerDbConnector, customerDbConnector)
+
```

Comment:
these methods could be shorter if they would make use of the class attributes. I would prefer a small refactoring so that it makes this part look as clean as the rest :)

Reply:
Yes, you are right. They would be shorter. On the other hand, we have to consider that the method signature gives insight into what resources are needed for what context. I would prefer to leave it as is. We could change the formatting, but I am convinced that it is good to have the create methods know what they require.

Reply:
Those lines should only change when you change a context creation in a major way. In such cases, it should be obvious what needs to change in the method calls.

Reply:
I can see your point, but I still think this blows up the methods more than it needs to. It also makes the readablility really hard. I would argue listing the context in the javadoc should be sufficient. Imagine how crazy big the methods will become if more features will be included that include multiple other features

Reply:
I would expect that the changes are still obvious even without the method parameter since the DM is now cleaned up

Reply:
f25c410